internal/services/dns/validate/dns_record_type_id.go

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package validate
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/dns/2023-07-01-preview/recordsets"
+)
+
+func ValidateRecordTypeID(recordType recordsets.RecordType) func(interface{}, string) ([]string, []error) {
+	return func(input interface{}, key string) (warnings []string, errors []error) {
+		warnings, errors = recordsets.ValidateRecordTypeID(input, key)
+		if len(warnings) > 0 || len(errors) > 0 {
+			return
+		}
+
+		// recordsets.ValidateRecordTypeID checks that input is a string and that it parses successfully so we don't need to recheck
+		parsed, _ := recordsets.ParseRecordTypeID(input.(string))
+		if parsed.RecordType != recordType {
+			errors = append(errors, fmt.Errorf("this resource only supports '%q' records", recordType))
+			return
+		}
+
+		return
+	}
+}
```

Comment:
Regarding the ID validation function here to ensure that the DNS record is of the correct type, this is absolutely correct way to handle the imports. 

My only remark is that for custom import function we tend to just attempt parsing the ID and returning an error if it fails or the resource type isn't what we're expecting instead of validating and then parsing since as you've pointed out in your comment, the validation asserts that the input is a string and then calls the parse function anyway, so IMO this could be simplified to 
```suggestion
func ValidateRecordTypeID(recordType recordsets.RecordType) func(interface{}, string) ([]string, []error) {
	return func(input interface{}, key string) (warnings []string, errors []error) {
		parsed, err := recordsets.ParseRecordTypeID(input.(string))
		if err != nil {
			errors = append(errors, err)
			return
		}
		if parsed.RecordType != recordType {
			errors = append(errors, fmt.Errorf("this resource only supports '%q' records", recordType))
			return
		}

		return
	}
}
```